dbms/src/Storages/Transaction/ApplySnapshot.cpp

```diff
@@ -139,6 +139,14 @@ void KVStore::checkAndApplyPreHandledSnapshot(const RegionPtrWrap & new_region,
     onSnapshot(new_region, old_region, old_applied_index, tmt);
 }
 
+std::pair<UInt64, bool> getTiFlashReplicaSyncInfo(StorageDeltaMergePtr & dm_storage)
+{
+    auto struct_lock = dm_storage->lockStructureForShare(getThreadNameAndID());
+    const auto & replica_info = dm_storage->getTableInfo().replica_info;
+    auto is_syncing = replica_info.count > 0 && replica_info.available && !(*replica_info.available);
```

Comment:
```suggestion
    auto is_syncing = replica_info.count > 0 && replica_info.available.has_value() && !(*replica_info.available);
```
this is more clear since `replica_info.available` is an `optional<bool>`